### PR TITLE
[14.0][IMP] edi_oca:  sync parent state on ack process

### PR DESCRIPTION
**Pull request by simgonzalez**
Mechanism to sync both EDI exchange together.
resolve (issue #789 )

### CODE REVIEW

**Review by gbrito — APPROVED** [review]

**Review by ecino — APPROVED** [review]

**Review by simahawk — CHANGES_REQUESTED** [review]
I'm not sure to understand the use case.
I'll comment on the linked issue.

File: edi_oca/models/edi_backend.py [hunk 1]
```diff
@@ -445,6 +445,10 @@ def _exchange_process_check(self, exchange_record):
             "input_processed_error",
         ]
 
+    def _update_related_ack(self, exchange_record):
```
Comment (simahawk) [inline edi_oca/models/edi_backend.py:448]:
```suggestion
    def _is_parent_update_needed(self, exchange_record):
```

File: edi_oca/models/edi_backend.py [hunk 2]
```diff
@@ -445,6 +445,10 @@ def _exchange_process_check(self, exchange_record):
...
+        if exchange_record.parent_id and exchange_record.type_id.ack_for_type_ids:
```
Comment (simahawk) [inline edi_oca/models/edi_backend.py:449]: I'm not sure the check on `ack_for_type_ids` is significant. What about this?
```suggestion
        if exchange_record.parent_id.ack_expected:
```

File: edi_oca/models/edi_backend.py [hunk 3]
```diff
@@ -456,6 +460,7 @@ def exchange_process(self, exchange_record):
         state = exchange_record.edi_exchange_state
         error = False
         message = None
+        origin_output_state = False
```
Comment (simahawk) [inline edi_oca/models/edi_backend.py:463]:
```suggestion
        parent_output_state = None
```

File: edi_oca/models/edi_backend.py [hunk 4]
```diff
@@ -464,20 +469,32 @@ def exchange_process(self, exchange_record):
...
             res = True
+            if self._update_related_ack(exchange_record):
+                origin_output_state = "output_sent_and_processed"
         finally:
             exchange_record.write(
                 {
                     "edi_exchange_state": state,
                     "exchange_error": error,
                     # FIXME: this should come from _compute_exchanged_on
+                    # TODO: shouldn't this update processed_on ?
```
Comment (simahawk) [inline edi_oca/models/edi_backend.py:486]: `exchanged_on` is a generic field. In this case, it reflects the time of processing. To see when it was "received" just check the create date.

**Review by simahawk — CHANGES_REQUESTED** [review]
Thanks for your proposal!
Beside the comments, here some more steps:
* tests
* squash commits
* rewrite commit msg. Something like this should work:
 
```
edi: sync parent state on ack process

$detailed_explanation_as_given on the linked issue
```

**Comment by ecino** [comment]
I will probably make a PR myself to address the comments.

**Pull request closed without merging**